sky/clouds/service_catalog/data_fetchers/fetch_gcp.py

```diff
@@ -518,6 +523,34 @@ def get_gpu_price(row: pd.Series, spot: bool) -> Optional[float]:
     return df
 
 
+def _get_tpu_response_for_zone(zone: str) -> list:
+    parent = f'projects/{project_id}/locations/{zone}'
+    # Sometimes the response is empty ({}) even for enabled zones. Here we
+    # retry the request for a few times.
+    backoff = common_utils.Backoff(initial_backoff=1)
+    for _ in range(TPU_RETRY_TIMES):
+        tpus_request = (
+            tpu_client.projects().locations().acceleratorTypes().list(
+                parent=parent))
+        try:
+            tpus_response = tpus_request.execute()
+            if tpus_response:
+                return tpus_response.get('acceleratorTypes', [])
```

Comment:
Should we retry for the case when `acceleratorTypes` does not exist? I found it fail sometimes for `us-central2-b` with no `acceleratorTypes` but succeed for other case

Reply:
Good point! Updated. Thanks!